sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogUtils.scala

```diff
@@ -162,6 +164,28 @@ object CatalogUtils {
     BucketSpec(numBuckets, normalizedBucketCols, normalizedSortCols)
   }
 
+  /**
+   * Convert URI to String.
+   * Since URI.toString does not decode for the uri string, we need to use
```

Comment:
`Since URI.toString does not decode the uri string, here we create a hadoop Path with the given URI, ad rely on hadoop Path.toString to decode the uri`